vault/resource_aws_secret_backend_static_role.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/util"
+)
+
+const (
+	awsStaticRolesAffix = "static-roles"
+)
+
+// awsStaticRolesRegex matches vault paths, capturing backend and role names. This is greedy-by-default,
+// which means if someone does something like 'a/static-roles/static-roles/b', the ambiguity will be
+// resolved by putting the extras into the backend part of the path.
+var awsStaticRolesRegex = regexp.MustCompile("^(.+)/static-roles/(.+)$")
+
+func awsSecretBackendStaticRoleResource() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		// backend is deprecated, but the other AWS resource types use it, and predate the deprecation.
+		// It's probably more helpful to the end user to maintain this consistency, in this particular case.
+		consts.FieldBackend: {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      consts.MountTypeAWS,
+			Description:  "The path where the AWS secrets backend is mounted.",
+			ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+		},
+		consts.FieldName: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Name of the role.",
+			ForceNew:    true,
+		},
+		consts.FieldUsername: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The username of the existing AWS IAM user to manage password rotation for.",
+			ForceNew:    true,
+		},
+		consts.FieldRotationPeriod: {
+			Type:        schema.TypeInt,
+			Required:    true,
+			Description: "How often Vault should rotate the password of the user entry.",
+		},
+	}
+	return &schema.Resource{
+		CreateContext: createUpdateAWSStaticRoleResource,
```

Comment:
We should add the `MountCreateContextWrapper` here, and ensure that this resource is only enabled on Vault versions >= 1.14. An example on how this is used can be found [here](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_ldap_secret_backend.go#L124)

Reply:
added!